src/app/components/header/header.component.ts

```diff
@@ -3,19 +3,28 @@ import {
   Component,
   ViewEncapsulation,
 } from '@angular/core';
-import { Observable } from 'rxjs';
-import { CategoryModel } from 'src/app/models/category.model';
-import { CategoriesService } from 'src/app/services/categories.service';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { CategoryModel } from '../../models/category.model';
+import { CategoriesService } from '../../services/categories.service';
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss'],
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class HeaderComponent {
   readonly categories$: Observable<CategoryModel[]> =
     this._categoriesService.getAllCategories();
+  private _showSideMenuSubject: BehaviorSubject<boolean> =
+    new BehaviorSubject<boolean>(false);
+  public showSideMenu$: Observable<boolean> =
+    this._showSideMenuSubject.asObservable();
 
   constructor(private _categoriesService: CategoriesService) {}
+
+  showSideMenu(showSideMenu: boolean): void {
+    this._showSideMenuSubject.next(showSideMenu);
```

Comment:
Maybe the better way is toggle value depends on prev? If prev was false, change for true but if prev was true, change for false